File: 7. Reverse Integer/Reverse Integer.rb
# @param {Integer} x
# @return {Integer}
def reverse(x)
    result = String.new
    negative = false
    if x < 0
    	negative = true
    	x = x * (-1)
    end

    i = 1

    if x < i 
    	return x
    else
    	while x >= i
    		num = x / i
    		result = result + num.to_s.split('').pop
    		i *= 10
    	end
    end
    
    result = result.to_i
    
    if (result > 2**31-1) || (result < -2**31)
		return 0
	end

    negative ?  result*(-1) : result
end

File: 7. Reverse Integer/Reverse Integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    if(x < 0) return -1 * reverse(x * -1);
    var result = parseInt(Number.prototype.toString.apply(x).split('').reverse().join(''),10)
    if(result > Math.pow(2,31) -1) result = 0;
    return result;
};
